cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/async/progress/OperationProgress.java

```diff
@@ -84,6 +86,21 @@ public synchronized String toString() {
     return sb.toString();
   }
 
+  public synchronized Map<String, Map<String, String>> getJsonStructure() {
+    Map<String, Map<String, String>> progressMap = new LinkedHashMap<>();
```

Comment:
While it is possible to build an ordered `json` string from a `LinkedHashMap` ([source](https://stackoverflow.com/questions/29491281/building-ordered-json-string-from-linkedhashmap)), I believe for the sake of processing the generated json file, it would be more preferable to make the step order more explicit rather than implicit -- i.e. hidden in the returned print order. Hence, I propose that we add a new field for each step that indicates the step number. We can name the field `step` and use the index of `_steps` as its value.